### PR TITLE
Fix IsTransparentProxy check in Microsoft.CSharp

### PR DESCRIPTION
**Pull request by stephentoub**
Fixes https://github.com/dotnet/coreclr/issues/1425

Thanks to @JamesNK for finding the issue and @mikedn for spotting the problem.

@AlekseyTs, @VSadov, there are no tests for Microsoft.CSharp in the repo.  Can you help with whatever verification needs to be done on this?

### CODE REVIEW

**Comment by mikedn** [comment]
@stephentoub  Thanks for making a fix, I was thinking to do it myself but I was sure that discussions about tests will start up :smile:

**Comment by stephentoub** [comment]
:smile:

File: src/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/BinderHelper.cs [hunk 1]
```diff
@@ -195,7 +195,10 @@ internal static bool IsWindowsRuntimeObject(DynamicMetaObject obj)
 
         private static bool IsTransparentProxy(object obj)
         {
-            return obj != null;
```
Comment (VSadov) [inline src/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/BinderHelper.cs]: This looks like just a logical bug in trying to keep the part of the check unaffected by absence of RemotingServices.IsTransparentProxy

Comment (stephentoub) [inline src/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/BinderHelper.cs]: Yes. You ok with the fix? Any additional validation needed?

**Comment by VSadov** [comment]
System.Dynamic.Runtime has a bunch of tests for C# dynamic. Perhaps not the most intuitive place, but that is where they are now.
I think the change is good and testing seems sufficient.

**Pull request merged**